compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -294,7 +286,7 @@ impl<'tcx> ConstToPat<'tcx> {
                 // Since we are behind a reference, we can just bubble the error up so we get a
                 // constant at reference type, making it easy to let the fallback call
                 // `PartialEq::eq` on it.
-                return Err(fallback_to_const_ref(self));
+                return Err(FallbackToConstRef);
```

Comment:
This is exactly the same behaviour as on master. We're behind a reference already, so we aren't even using the new code path in `test.rs`.